src/mds/MDCache.cc

```diff
@@ -8813,7 +8813,8 @@ void MDCache::open_ino(inodeno_t ino, int64_t pool, MDSInternalContextBase* fin,
 void MDCache::find_ino_peers(inodeno_t ino, MDSInternalContextBase *c, mds_rank_t hint)
 {
   dout(5) << "find_ino_peers " << ino << " hint " << hint << dendl;
-  assert(!have_inode(ino));
+  CInode *in = get_inode(ino);
+  assert(!have_inode(ino) || (in && in->state_test(CInode::STATE_PURGING)));
```

Comment:
should be:

if (in && in->state_test(CInode::STATE_PURGING)) {
  c->complete(-ESTALE);
  return;
}
assert(!in)

Reply:
good suggestion!